actionpack/lib/action_dispatch/journey/route.rb

```diff
@@ -70,24 +70,23 @@ def initialize(name:, app: nil, path:, constraints: {}, required_defaults: [], d
         @path_formatter    = @path.build_formatter
         @scope_options     = scope_options
         @internal          = internal
+
+        @path.ast.route = self
       end
 
       def eager_load!
         path.eager_load!
-        ast
         parts
         required_defaults
         nil
       end
 
       def ast
-        @decorated_ast ||= begin
-          decorated_ast = path.ast
-          decorated_ast.find_all(&:terminal?).each { |n| n.memo = self }
-          # inject any regexp requirements for `star` nodes so they can be determined nullable, which requires knowing if the regex accepts an empty string.
-          decorated_ast.find_all(&:star?).each { |n| n.regexp = path.requirements[n.name.to_sym] unless path.requirements[n.name.to_sym].nil? }
-          decorated_ast
-        end
+        path.ast
+      end
+
+      def ast_root
+        @ast_root ||= path.ast.root
```

Comment:
Why the memoization here? Neither `ast` or `root` are calculated.

Reply:
It's there because we wipe out path.ast after eager_loading (see https://github.com/rails/rails/pull/39935/files#diff-b5197cabb760000bb11b71067d918d9b5c246c42a9b47563ee0e7b221add7bfcR31). Probably makes more sense to set this in initialize and use an attr_reader. That'll get rid of the awkwardness of needing to call this method prior to eager loading.

Reply:
Right. I think that would be better.

Reply:
Done